api/src/wfl/module/covid.clj

```diff
@@ -303,8 +307,50 @@
         (assoc :type terra-workspace-sink-type))
     (throw (ex-info "Invalid sink_items" details))))
 
+(defn ^:visible-for-testing rename-gather
+  "Transform the `values` using the transformation defined in `mapping`."
+  [values mapping]
+  (letfn [(literal? [x] (str/starts-with? x "$"))
+          (go! [v]
+            (cond (literal? v) (subs v 1 (count v))
+                  (string?  v) (values (keyword v))
+                  (map?     v) (rename-gather values v)
+                  (coll?    v) (keep go! v)
+                  :else        (throw (ex-info "Unknown operation"
+                                               {:operation v}))))]
+    (into {} (for [[k v] mapping] [k (go! v)]))))
+
+(defn ^:private terra-workspace-sink-to-attributes
+  [{:keys [outputs] :as workflow} fromOutputs]
+  (when-not (map? fromOutputs)
+    (throw (IllegalStateException. "fromOutputs is malformed")))
+  (try
+    (rename-gather outputs fromOutputs)
+    (catch Exception cause
+      (throw (ex-info "Failed to coerce workflow outputs to attribute values"
+                      {:fromOutputs fromOutputs :workflow workflow}
+                      cause)))))
+
```

Comment:
Could we have an `:as _sink` on the second parameter below?

Reply:
sure